packages/devextreme/js/__internal/ui/stepper/stepper.ts

```diff
@@ -101,36 +101,51 @@ class Stepper extends CollectionWidgetAsync<StepperProperties> {
     return icon;
   }
 
+  _getStepLabel(data: Item): dxElementWrapper {
+    const { title, optional } = data;
+    const hasTitle = isDefined(title);
+
+    if (!hasTitle && !optional) {
+      return $();
+    }
+
+    const $stepLabel = $('<div>').addClass(STEP_LABEL_CLASS);
+
+    if (hasTitle) {
+      const $stepTitle = $('<div>')
```

Comment:
Let's move the logic with getting title and optional elements to separate methods